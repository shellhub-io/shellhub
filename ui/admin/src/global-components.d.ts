// src/global-components.d.ts
import SnackbarComponent from "./components/Snackbar/Snackbar.vue";

declare module "@vue/runtime-core" {
  export interface GlobalComponents {
    SnackbarComponent: typeof SnackbarComponent;
    RouterLink: typeof import("vue-router")["RouterLink"];
    RouterView: typeof import("vue-router")["RouterView"];
  }
}
