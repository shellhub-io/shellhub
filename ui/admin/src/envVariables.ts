/* eslint-disable import/prefer-default-export */
export const envVariables = {
  announcementsEnable: process.env.SHELLHUB_ANNOUNCEMENTS === "true",
  tinyMceKey: process.env.TINYMCE_API_KEY,
};
