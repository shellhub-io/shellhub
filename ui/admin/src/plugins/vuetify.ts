// Styles
// import "@mdi/font/css/materialdesignicons.css";
import "vuetify/styles";
import { VNumberInput } from "vuetify/labs/VNumberInput";

// Vuetify
import { createVuetify } from "vuetify";

export default createVuetify({
  theme: {
    defaultTheme: "dark",
  },
  components: {
    VNumberInput,
  },
});
