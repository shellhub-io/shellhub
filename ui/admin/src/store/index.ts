import { InjectionKey } from "vue";
import { createStore, Store, useStore as vuexUseStore } from "vuex";
import { stats, StatsState } from "./modules/stats";

import { UsersState, users } from "./modules/users";
import { AuthState, auth } from "./modules/auth";
import { DevicesState, devices } from "./modules/devices";
import { firewallRules, FirewallRulesState } from "./modules/firewall_rules";
import { namespaces, NamespacesState } from "./modules/namespaces";
import { layout, LayoutState } from "./modules/layout";
import { license, LicenseState } from "./modules/license";
import { publicKeys, PublicKeysState } from "./modules/public_keys";
import { sessions, SessionsState } from "./modules/sessions";
import { snackbar, SnackbarState } from "./modules/snackbar";
import { spinner, SpinnerState } from "./modules/spinner";
import { announcement, AnnouncementState } from "./modules/announcement";
import { instance, InstanceState } from "./modules/instance";

export interface State {
  user: UsersState;
  auth: AuthState;
  devices: DevicesState;
  firewallRules: FirewallRulesState;
  layout: LayoutState;
  license: LicenseState;
  namespaces: NamespacesState;
  publicKeys: PublicKeysState;
  sessions: SessionsState;
  snackbar: SnackbarState;
  spinner: SpinnerState;
  stats: StatsState;
  announcement: AnnouncementState;
  instance: InstanceState;
}

// eslint-disable-next-line symbol-description
export const key: InjectionKey<Store<State>> = Symbol();

export const store = createStore<State>({
  modules: {
    users,
    auth,
    devices,
    firewallRules,
    layout,
    license,
    namespaces,
    publicKeys,
    sessions,
    snackbar,
    spinner,
    stats,
    announcement,
    instance,
  },
});

export function useStore(): Store<State> {
  return vuexUseStore(key);
}
