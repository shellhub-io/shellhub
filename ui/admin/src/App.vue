<template>
  <v-app>
    <component :is="layout" :data-test="`${layout}-component`" />
  </v-app>
</template>

<script lang="ts">
import { defineComponent, onMounted, computed } from "vue";
import { useRouter } from "vue-router";
import SimpleLayout from "./layouts/SimpleLayout.vue";
import AppLayout from "./layouts/AppLayout.vue";
import { useStore } from "./store";
import { INotificationsError } from "./interfaces/INotifications";

export default defineComponent({
  name: "App",
  components: {
    appLayout: AppLayout,
    simpleLayout: SimpleLayout,
  },
  setup() {
    const store = useStore();
    const router = useRouter();

    const layout = computed(() => store.getters["layout/getLayout"]);

    const isLoggedIn = computed(() => store.getters["auth/isLoggedIn"]);

    const currentRoute = computed(() => router.currentRoute.value.path);

    onMounted(async () => {
      if (!isLoggedIn.value) {
        try {
          await store.dispatch("auth/logout");
          store.dispatch("layout/setLayout", "simpleLayout");
          router.push("/login");
        } catch {
          store.dispatch("snackbar/showSnackbarErrorAction", INotificationsError.namespaceLoad);
        }
      }

      if (isLoggedIn.value && currentRoute.value !== "/login") {
        const license = await store.dispatch("license/get");
        if (!license || license.expired) {
          store.dispatch("snackbar/showSnackbarErrorAction", INotificationsError.license);
          store.dispatch("layout/setLayout", "appLayout");
          router.push("/license");
        }
        store.dispatch("layout/setLayout", "appLayout");
      }
    });

    return {
      layout,
    };
  },
});
</script>
