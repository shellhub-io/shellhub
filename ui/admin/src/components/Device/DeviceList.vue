<template>
  <div>
    <DataTable
      :headers="headers"
      :items="devices"
      :itemsPerPage="itemsPerPage"
      :nextPage="next"
      :previousPage="prev"
      :loading="loading"
      :totalCount="numberDevices"
      :actualPage="page"
      :comboboxOptions="[10, 20, 50, 100]"
      @changeItemsPerPage="changeItemsPerPage"
      @clickNextPage="next"
      @clickPreviousPage="prev"
      @clickSortableIcon="sortByItem"
      data-test="devices-list"
    >
      <template v-slot:rows>
        <tr v-for="(item, i) in devices" :key="i">
          <td>
            <v-icon v-if="item.online" color="success" data-test="sucess-icon">
              mdi-check-circle
            </v-icon>
            <v-icon v-else color="#E53935" data-test="error-icon"> mdi-close-circle </v-icon>
          </td>
          <td>{{ item.name }}</td>
          <td class="d-flex align-center">
            <DeviceIcon :icon="item.info.id" class="mr-2" />
            {{ item.info.prettyName }}
          </td>
          <td>
            <span
              @click="goToNamespace(item.tenant_id)"
              @keypress.enter="goToNamespace(item.tenant_id)"
              tabindex="0"
              class="hover"
            >
              {{ item.namespace }}
            </span>
          </td>
          <td>
            <div v-if="item.tags[0]">
              <v-tooltip
                v-for="(tag, index) in item.tags"
                :key="index"
                bottom
                :disabled="!showTag(tag)"
              >
                <template #activator="{ props }">
                  <v-chip size="small" v-bind="props" v-on="props">
                    {{ displayOnlyTenCharacters(tag) }}
                  </v-chip>
                </template>

                <span v-if="showTag(tag)">
                  {{ tag }}
                </span>
              </v-tooltip>
            </div>
          </td>
          <td>
            {{ formatDate(item.last_seen) }}
          </td>
          <td>
            <v-chip size="small">
              {{ item.status }}
            </v-chip>
          </td>
          <td>
            <v-tooltip bottom anchor="bottom">
              <template v-slot:activator="{ props }">
                <v-icon
                  tag="a"
                  dark
                  v-bind="props"
                  @click="redirectToDevice(item.uid)"
                  @keypress.enter="redirectToDevice(item.uid)"
                  tabindex="0"
                >mdi-information
                </v-icon>
              </template>
              <span>Info</span>
            </v-tooltip>
          </td>
        </tr>
      </template>
    </DataTable>
  </div>
</template>

<script lang="ts">
import { defineComponent, ref, onMounted, watch, computed } from "vue";
import { useRouter } from "vue-router";
import { useStore } from "../../store";
import DataTable from "../DataTable.vue";
import DeviceIcon from "./DeviceIcon.vue";
import { formatDate } from "../../hooks/formateDate";
import displayOnlyTenCharacters from "../../hooks/string";
import showTag from "../../hooks/tag";
import { INotificationsError } from "../../interfaces/INotifications";

export default defineComponent({
  setup() {
    const store = useStore();
    const router = useRouter();
    const loading = ref(false);
    const filter = ref("");
    const itemsPerPage = ref(10);
    const page = ref(1);

    const devices = computed(() => store.getters["devices/list"]);
    const numberDevices = computed(() => store.getters["devices/numberDevices"]);

    onMounted(async () => {
      try {
        loading.value = true;
        await store.dispatch("devices/fetch", {
          perPage: itemsPerPage.value,
          page: 1,
          filter: "",
          sortStatusField: "",
          sortStatusString: "",
        });
      } catch {
        store.dispatch("snackbar/showSnackbarErrorAction", INotificationsError.deviceList);
      } finally {
        loading.value = false;
      }
    });

    const getDevices = async (perPagaeValue: number, pageValue: number) => {
      try {
        loading.value = true;

        const hasDevices = await store.dispatch("devices/fetch", {
          perPage: perPagaeValue,
          page: pageValue,
          filter: filter.value,
          sortStatusField: store.getters["devices/sortStatusField"],
          sortStatusString: store.getters["devices/sortStatusString"],
        });

        if (!hasDevices) {
          page.value--;
        }

        loading.value = false;
      } catch (error) {
        store.dispatch("snackbar/showSnackbarErrorAction", INotificationsError.deviceList);
      }
    };

    const sortByItem = async (field: string) => {
      let sortStatusString = store.getters["devices/sortStatusString"];
      if (sortStatusString === "") {
        sortStatusString = "asc";
      } else if (sortStatusString === "asc") {
        sortStatusString = "desc";
      } else {
        sortStatusString = "";
      }
      await store.dispatch("devices/setSortStatus", {
        sortStatusField: field,
        sortStatusString,
      });
      await getDevices(itemsPerPage.value, page.value);
    };

    const next = async () => {
      await getDevices(itemsPerPage.value, ++page.value);
    };

    const prev = async () => {
      try {
        if (page.value > 1) await getDevices(itemsPerPage.value, --page.value);
      } catch (error) {
        store.dispatch("snackbar/setSnackbarErrorDefault");
      }
    };

    const changeItemsPerPage = async (newItemsPerPage: number) => {
      itemsPerPage.value = newItemsPerPage;
    };

    watch(itemsPerPage, async () => {
      await getDevices(itemsPerPage.value, page.value);
    });

    const goToNamespace = (namespace: string) => {
      router.push({ name: "namespaceDetails", params: { id: namespace } });
    };

    const redirectToDevice = (deviceId: string) => {
      router.push({ name: "deviceDetails", params: { id: deviceId } });
    };

    return {
      headers: [
        {
          text: "Online",
          value: "online",
          sortable: true,
        },
        {
          text: "Hostname",
          value: "name",
          sortable: true,
        },
        {
          text: "Info",
          value: "info",
          sortable: true,
        },
        {
          text: "Namespace",
          value: "namespace",
          sortable: true,
        },
        {
          text: "Tags",
          value: "tags",
        },
        {
          text: "Last Seen",
          value: "last_seen",
          sortable: true,
          align: "center",
        },
        {
          text: "Status",
          value: "status",
          sortable: true,
        },
        {
          text: "Actions",
          value: "actions",
        },
      ],
      itemsPerPage,
      page,
      loading,
      devices,
      numberDevices,
      next,
      prev,
      sortByItem,
      showTag,
      displayOnlyTenCharacters,
      formatDate,
      goToNamespace,
      redirectToDevice,
      changeItemsPerPage,
    };
  },
  components: { DataTable, DeviceIcon },
});
</script>

<style scoped>
.hover:hover,
.hover:focus {
  cursor: pointer;
  text-decoration: underline;
}
</style>
