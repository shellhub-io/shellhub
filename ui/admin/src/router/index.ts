import { createRouter, createWebHistory } from "vue-router";
import Login from "@admin/views/Login.vue";
import Dashboard from "@admin/views/Dashboard.vue";
import Users from "@admin/views/Users.vue";
import SettingsLicense from "@admin/components/Settings/SettingsLicense.vue";
import SettingsAuthentication from "@admin/components/Settings/SettingsAuthentication.vue";
import Namespaces from "@admin/views/Namespaces.vue";
import Settings from "@admin/views/Settings.vue";

import { store } from "../store";

const routes = [
  {
    path: "/login",
    name: "login",
    component: Login,
  },
  {
    path: "/",
    name: "dashboard",
    component: Dashboard,
  },
  {
    path: "/users",
    name: "users",
    component: Users,
  },
  {
    path: "/user/:id",
    name: "userDetails",
    component: () => import("@admin/views/UserDetails.vue"),
  },
  {
    path: "/devices",
    name: "devices",
    component: () => import("@admin/views/Device.vue"),
    redirect: {
      name: "listDevices",
    },
    children: [
      {
        path: "",
        name: "listDevices",
        component: () => import("@admin/views/Device.vue"),
      },
    ],
  },
  {
    path: "/device/:id",
    name: "deviceDetails",
    component: () => import("@admin/views/DeviceDetails.vue"),
  },
  {
    path: "/sessions",
    name: "sessions",
    component: () => import("@admin/views/Sessions.vue"),
  },
  {
    path: "/session/:id",
    name: "sessionDetails",
    component: () => import("@admin/views/SessionDetails.vue"),
  },
  {
    path: "/settings",
    name: "Settings",
    component: Settings,
    redirect: { name: "SettingProfile" },
    children: [
      {
        path: "authentication",
        name: "SettingAuthentication",
        component: SettingsAuthentication,
      },
      {
        path: "license",
        name: "SettingLicense",
        component: SettingsLicense,
      },
    ],
  },
  {
    path: "/firewall-rules",
    name: "firewall-rules",
    component: () => import("@admin/views/FirewallRules.vue"),
  },
  {
    path: "/firewall-rules/:id",
    name: "firewallRulesDetails",
    component: () => import("@admin/views/FirewallRulesDetails.vue"),
  },
  {
    path: "/namespaces",
    name: "namespaces",
    component: Namespaces,
  },
  {
    path: "/namespace/:id",
    name: "namespaceDetails",
    component: () => import("@admin/views/NamespaceDetails.vue"),
  },
  {
    path: "/announcements",
    name: "announcements",
    component: () => import("@admin/views/Announcements.vue"),
  },
  {
    path: "/announcement/:uuid",
    name: "announcementDetails",
    component: () => import("@admin/views/AnnouncementDetails.vue"),
  },
  {
    path: "/new-announcement",
    name: "new-announcement",
    component: () => import("@admin/views/NewAnnouncement.vue"),
  },
];

const router = createRouter({
  history: createWebHistory("/admin/"),
  routes,
});

router.beforeEach((to, from, next) => {
  if (to.path !== "/login") {
    if (store.getters["auth/isLoggedIn"]) {
      return next();
    }
    return next(`/login?redirect=${to.path}`);
  }
  if (store.getters["auth/isLoggedIn"]) {
    if (to.path === "/login" && to.query.token) {
      return next();
    }
    return next("/");
  }

  return next();
});

export default router;
