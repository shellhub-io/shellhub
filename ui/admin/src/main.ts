import { createApp } from "vue";
import vuetify from "@ui/plugins/vuetify";
import App from "./App.vue";
import { loadFonts } from "./plugins/webfontloader";
import { key, store } from "./store";
import router from "./router";
import SnackbarComponent from "./components/Snackbar/Snackbar.vue";

const app = createApp(App);

loadFonts();

app.use(vuetify);
app.use(router);
app.use(store, key);
app.component("SnackbarComponent", SnackbarComponent);
app.mount("#app");
