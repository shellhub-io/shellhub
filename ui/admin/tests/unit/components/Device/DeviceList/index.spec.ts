import { createStore } from "vuex";
import { createVuetify } from "vuetify";
import { mount, VueWrapper } from "@vue/test-utils";
import { beforeEach, describe, expect, it, vi } from "vitest";
import DeviceList from "../../../../../src/components/Device/DeviceList.vue";
import { key } from "../../../../../src/store";
import routes from "../../../../../src/router";

const headers = [
  { text: "Online", value: "online", sortable: true },
  { text: "Hostname", value: "name", sortable: true },
  { text: "Info", value: "info", sortable: true },
  { text: "Namespace", value: "namespace", sortable: true },
  { text: "Tags", value: "tags" },
  { text: "Last Seen", value: "last_seen", sortable: true, align: "center" },
  { text: "Status", value: "status", sortable: true },
  { text: "Actions", value: "actions" },
];

const devices = [
  {
    created_at: "2020-05-20T19:58:53.276Z",
    identity: { mac: "00:00:00:00:00:00" },
    info: {
      arch: "x86_64",
      id: "linuxmint",
      platform: "linuxmint",
      prettyName: "Linux Mint 19.3",
      version: "18.4.2",
    },
    last_seen: "2020-05-20T19:58:53.276Z",
    name: "tests",
    namespace: "dev",
    online: true,
    position: {
      latitude: 12,
      longitude: 12,
    },
    public_key: "xxxxxxxxxxxxxxxx",
    remote_addr: "127.0.0.1",
    status: "accepted",
    tags: ["xxxx", "yyyyy"],
    tenant_id: "00000000",
    uid: "a582b47a42d",
  },
];

const store = createStore({
  state: {
    devices,
  },
  getters: {
    "devices/list": (state) => state.devices,
    "devices/numberDevices": (state) => state.devices.length,
    "auth/isLoggedIn": () => false,
  },
  actions: {
    "modals/showAddDevice": vi.fn(),
    "devices/fetch": vi.fn(),
    "devices/rename": vi.fn(),
    "devices/resetListDevices": vi.fn(),
    "stats/get": vi.fn(),
  },
});

describe("Device List", () => {
  let wrapper: VueWrapper<any>;

  beforeEach(() => {
    const vuetify = createVuetify();

    wrapper = mount(DeviceList, {
      global: {
        plugins: [[store, key], vuetify, routes],
      },
    });
  });

  it("Is a Vue instance", () => {
    expect(wrapper).toBeTruthy();
  });

  it("Renders the component", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("Renders the template with data", () => {
    const dt = wrapper.find("[data-test]");
    expect(dt.attributes()["data-test"]).toBe("devices-list");
    expect(wrapper.vm.headers).toEqual(headers);
    expect(wrapper.vm.devices).toEqual(devices);
    expect(wrapper.vm.loading).toEqual(false);
    expect(wrapper.vm.itemsPerPage).toEqual(10);
  });

  it("Renders data in the computed", async () => {
    const devices = await wrapper.vm.getListDevices;
    expect(devices).toEqual(devices);
  });
});
