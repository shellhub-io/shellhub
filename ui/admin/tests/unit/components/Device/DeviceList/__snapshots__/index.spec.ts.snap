// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Device List > Renders the component 1`] = `
"<div data-v-aca44803="">
  <div data-v-90b5158f="" data-v-aca44803="" data-test="devices-list">
    <div data-v-90b5158f="" class="v-table v-theme--light v-table--density-default">
      <!---->
      <div class="v-table__wrapper">
        <table>
          <thead data-v-90b5158f="">
            <tr data-v-90b5158f="">
              <th data-v-90b5158f="" class="text-left"><span data-v-90b5158f="" tabindex="0" class="hover" aria-describedby="v-tooltip-0">Online <!----><!--teleport start--><!--teleport end--></span></th>
              <th data-v-90b5158f="" class="text-left"><span data-v-90b5158f="" tabindex="0" class="hover" aria-describedby="v-tooltip-1">Hostname <!----><!--teleport start--><!--teleport end--></span></th>
              <th data-v-90b5158f="" class="text-left"><span data-v-90b5158f="" tabindex="0" class="hover" aria-describedby="v-tooltip-2">Info <!----><!--teleport start--><!--teleport end--></span></th>
              <th data-v-90b5158f="" class="text-left"><span data-v-90b5158f="" tabindex="0" class="hover" aria-describedby="v-tooltip-3">Namespace <!----><!--teleport start--><!--teleport end--></span></th>
              <th data-v-90b5158f="" class="text-left"><span data-v-90b5158f="">Tags</span></th>
              <th data-v-90b5158f="" class="text-center"><span data-v-90b5158f="" tabindex="0" class="hover" aria-describedby="v-tooltip-4">Last Seen <!----><!--teleport start--><!--teleport end--></span></th>
              <th data-v-90b5158f="" class="text-left"><span data-v-90b5158f="" tabindex="0" class="hover" aria-describedby="v-tooltip-5">Status <!----><!--teleport start--><!--teleport end--></span></th>
              <th data-v-90b5158f="" class="text-left"><span data-v-90b5158f="">Actions</span></th>
            </tr>
          </thead>
          <tbody data-v-90b5158f="">
            <tr data-v-aca44803="">
              <td data-v-aca44803=""><i data-v-aca44803="" class="mdi-check-circle mdi v-icon notranslate v-theme--light v-icon--size-default text-success" aria-hidden="true" data-test="sucess-icon"></i></td>
              <td data-v-aca44803="">tests</td>
              <td data-v-aca44803="" class="d-flex align-center"><i data-v-aca44803="" class="fl-linuxmint mr-2" style="font-size: 20px;"></i> Linux Mint 19.3</td>
              <td data-v-aca44803=""><span data-v-aca44803="" tabindex="0" class="hover">dev</span></td>
              <td data-v-aca44803="">
                <div data-v-aca44803=""><span data-v-aca44803="" class="v-chip v-theme--light v-chip--density-default v-chip--size-small v-chip--variant-tonal" draggable="false" aria-describedby="v-tooltip-6"><!----><span class="v-chip__underlay"></span>
                  <!---->
                  <!---->
                  <div class="v-chip__content" data-no-activator="">xxxx</div>
                  <!---->
                  <!----></span>
                  <!--teleport start-->
                  <!--teleport end--><span data-v-aca44803="" class="v-chip v-theme--light v-chip--density-default v-chip--size-small v-chip--variant-tonal" draggable="false" aria-describedby="v-tooltip-8"><!----><span class="v-chip__underlay"></span>
                  <!---->
                  <!---->
                  <div class="v-chip__content" data-no-activator="">yyyyy</div>
                  <!---->
                  <!----></span>
                  <!--teleport start-->
                  <!--teleport end-->
                </div>
              </td>
              <td data-v-aca44803="">Wednesday, May 20th 2020, 7:58:53 pm</td>
              <td data-v-aca44803=""><span data-v-aca44803="" class="v-chip v-theme--light v-chip--density-default v-chip--size-small v-chip--variant-tonal" draggable="false"><!----><span class="v-chip__underlay"></span>
                <!---->
                <!---->
                <div class="v-chip__content" data-no-activator="">accepted</div>
                <!---->
                <!----></span>
              </td>
              <td data-v-aca44803=""><a data-v-aca44803="" class="mdi-information mdi v-icon notranslate v-theme--light v-icon--size-default v-icon--clickable" role="button" aria-hidden="false" tabindex="0" dark="" aria-describedby="v-tooltip-11"></a>
                <!--teleport start-->
                <!--teleport end-->
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <!---->
    </div>
    <hr data-v-90b5158f="" class="v-divider v-theme--light" aria-orientation="horizontal" role="separator">
    <!--v-if-->
    <div data-v-90b5158f="" class="d-flex w-100 justify-end align-center"><span data-v-90b5158f="" class="text-subtitle-2 mr-4">Items per page:</span>
      <div data-v-90b5158f="">
        <div data-v-90b5158f="" class="v-input v-input--horizontal v-input--density-default v-theme--light v-locale--is-ltr v-input--dirty v-text-field v-input--plain-underlined v-combobox v-combobox--single mb-4">
          <!---->
          <div class="v-input__control">
            <div class="v-field v-field--active v-field--appended v-field--dirty v-field--no-label v-field--variant-underlined v-theme--light v-locale--is-ltr" role="combobox" aria-haspopup="menu" aria-expanded="false" aria-controls="v-menu-14">
              <div class="v-field__overlay"></div>
              <div class="v-field__loader">
                <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
                  <!---->
                  <div class="v-progress-linear__background"></div>
                  <div class="v-progress-linear__buffer" style="width: 0%;"></div>
                  <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
                    <div class="v-progress-linear__indeterminate">
                      <div class="v-progress-linear__indeterminate long"></div>
                      <div class="v-progress-linear__indeterminate short"></div>
                    </div>
                  </transition-stub>
                  <!---->
                </div>
              </div>
              <!---->
              <div class="v-field__field" data-no-activator="">
                <!---->
                <!---->
                <!---->
                <div class="v-field__input" data-no-activator="">
                  <!---->
                  <!---->
                  <div class="v-combobox__selection"><span class="v-combobox__selection-text">10<!----></span></div><input size="1" type="text" id="input-12" aria-describedby="input-12-messages" outlined="" value="10">
                </div>
                <!---->
              </div>
              <!---->
              <div class="v-field__append-inner">
                <!----><i class="mdi-menu-down mdi v-icon notranslate v-theme--light v-icon--size-default v-icon--clickable v-combobox__menu-icon" role="button" aria-hidden="false" tabindex="-1" aria-label="Open" title="Open"></i>
                <!---->
              </div>
              <div class="v-field__outline">
                <!---->
                <!---->
              </div>
            </div>
          </div>
          <!---->
          <!---->
        </div>
      </div>
      <div data-v-90b5158f="" class="d-flex align-center"><button data-v-90b5158f="" type="button" class="v-btn v-btn--disabled v-btn--icon v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-plain" disabled=""><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
          <!----><span class="v-btn__content" data-no-activator=""><i class="mdi-chevron-left mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true"></i></span>
          <!---->
          <!---->
        </button><span data-v-90b5158f="" class="text-subtitle-2">1 of 1</span><button data-v-90b5158f="" type="button" class="v-btn v-btn--disabled v-btn--icon v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-plain" disabled=""><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
          <!----><span class="v-btn__content" data-no-activator=""><i class="mdi-chevron-right mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true"></i></span>
          <!---->
          <!---->
        </button></div>
    </div>
  </div>
</div>"
`;
